pkg/ignition/builder.go

```diff
@@ -99,10 +101,11 @@ func (b *ignitionBuilder) Generate() ([]byte, error) {
 		"/etc/NetworkManager/conf.d/clientid.conf",
 		0644, false,
 		[]byte("[connection]\nipv6.dhcp-duid=ll\nipv6.dhcp-iaid=mac")))
+
 	config.Storage.Files = append(config.Storage.Files, ignitionFileEmbed(
 		"/etc/NetworkManager/dispatcher.d/01-hostname",
 		0744, false,
-		[]byte("[[ \"$DHCP6_FQDN_FQDN\" =~ \".\" ]] && hostnamectl set-hostname --static --transient $DHCP6_FQDN_FQDN")))
+		[]byte("[[ \"$DHCP6_FQDN_FQDN\" =~ \".\" ]] && hostnamectl set-hostname --static --transient $DHCP6_FQDN_FQDN; [[ $HOSTNAME = localhost ]] && hostnamectl set-hostname --transient b.hostname")))
```

Comment:
Just for my education: what is `b.hostname` here? How does it work?

Reply:
During the installer bootstrap a bunch of files are created containing the nmstate configuration defined in the `install-config.yaml` file (per host) (see [here](https://github.com/openshift/installer/blob/303a3c7adcc718f48e1aa372acd92f31b4685642/data/data/bootstrap/baremetal/files/usr/local/bin/startironic.sh.template#L174)) and then passed to the icc. These file are named as the hostname specified in the config. Here each filename is extracted and passed in this `ignitionBuilder` struct as a field, to be used as hostname).

Reply:
So, it will execute `hostnamectl set-hostname --transient b.hostname` right? won't it set the hostnames of all nodes to literal string `b.hostname`?

Also I guess hostnamectl is different in coreos? mine has `hostnamectl hostname`, at least according to the man page.

Reply:
I think we should be sprintfing the variable `b.hostname` in here, not setting the hostname to the literal string `"b.hostname"` though, right?

Reply:
yeah, so
```go
fmt.Sprintf("[[ \"$DHCP6_FQDN_FQDN\" =~ \".\" ]] && hostnamectl set-hostname --static --transient $DHCP6_FQDN_FQDN; [[ $HOSTNAME = localhost ]] && hostnamectl set-hostname --transient %s", b.hostname)
```